htlcswitch/link.go

```diff
@@ -1525,10 +1537,22 @@ func (l *channelLink) handleDownStreamPkt(pkt *htlcPacket, isReProcess bool) {
 		htlc.ChanID = l.ChanID()
 		htlc.ID = pkt.incomingHTLCID
 
-		// Finally, we send the HTLC message to the peer which
-		// initially created the HTLC.
+		// We send the HTLC message to the peer which initially created
+		// the HTLC.
 		l.cfg.Peer.SendMessage(false, htlc)
 		isSettle = true
+
+		// Finally, we notify the failure. If the packet does not have a
+		// link failure set, it failed further down the route so we
+		// notify a forwarding failure. Otherwise, we notify a link
+		// failure because it failed at our node.
+		if pkt.linkFailure != nil {
+			l.cfg.HtlcNotifier.NotifyOutgoingLinkFailEvent(
+				pkt, lntypes.Hash{},
```

Comment:
Empty hash?

Reply:
Needed to circle back to this, I wasn't sure if `pkt.circuit` would always be non-nil so that we could use `pkt.circuit.PaymentHash` here. Will investigate and update.